jenkins-shared-lib/vars/README.md

```diff
@@ -0,0 +1,134 @@
+# Jenkins Shared Library
+
+This directory contains reusable Groovy scripts for Jenkins pipelines. These scripts can be used to perform common tasks such as checking out Git repositories, building code, building Docker images and pushing Docker images.
+
+## Scripts
+
+### 1. `checkoutGitRepo.groovy`
+This script checks out a Git repository from a specified URL and branch.
+
+#### Usage
+```groovy
+checkoutGitRepo(String repoUrl, String branch = 'main')
+```
+#### Parameters
+`repoUrl`: The URL of the Git repository.
+`branch`: The branch to checkout (default is main).
+
+Example
+
+```groovy
+checkoutGitRepo('https://github.com/username/repo.git', 'main')
+```
+
+### 2. `buildCode.groovy`
+This script is used to build code using a specified build command.
+
+#### Usage
+```groovy
+buildCode(String buildCommand)
+```
+#### Parameters
+`buildCommand`: The command to build the code (e.g., mvn clean install, npm install).
+
+Example
+
+```groovy
+buildCode('mvn clean install')
+```
+### 3. `buildDockerImage.groovy`
+This script builds a Docker image using a specified Dockerfile path, image name, and image tag.
+
+#### Usage
+```groovy
+buildDockerImage(String dockerfilePath, String imageName, String imageTag)
+```
+#### Parameters
+`dockerfilePath`: The path to the Dockerfile.
+`imageName`: The name of the Docker image.
+`imageTag`: The tag for the Docker image.
+
+Example
+
+```groovy
+buildDockerImage('Dockerfile', 'sample', 'latest')
+```
+
+### 4. `pushDockerImage.groovy`
+This script tags and pushes a Docker image to a specified Docker repository URL.
+
+#### Usage
+```groovy
+pushDockerImage(String imageName, String imageTag, String dockerRepoUrl)
+```
+#### Parameters
+`imageName`: The name of the Docker image.
+`imageTag`: The tag of the Docker image.
+`dockerRepoUrl`: The URL of the Docker repository.
+
+Example
+
+```groovy
+pushDockerImage('sample', 'latest', 'username/repo')
+```
+
+##### Example Jenkins Pipeline
+Here is an example of how to use these scripts in a Jenkins pipeline where I used a public repo which contains a Spring Boot based Java web application:
+
```

Comment:
@Rithin-QB ,

Thank you , but overall this is of almost no use. Ideally what we would like to see is the entire pipeline setup via a shared library.

So we can call
```
groovy
@Library(...)
def env = params.env
pipeline{
    env = env
    git_repo = xyz
etc....
}
```

and that will in turn automatically run the entire pipeline. If you really want you can utilize further fucntions for the various stages within the same, but only use functions for things that are complex would be my suggestion. If it's going to be a one line function it makes no sense in that case.

Reply:
I have restructured the shared library to encapsulate the entire pipeline within a single script. Now, the pipeline setup can be easily managed and invoked via the shared library with the ability to customize environment parameters.